src/platform/backends/shared/base_virtual_machine.cpp

```diff
@@ -17,30 +17,44 @@
 
 #include "base_virtual_machine.h"
 
+#include <multipass/logging/log.h>
+
+namespace mp = multipass;
+namespace mpl = multipass::logging;
+
 namespace multipass
 {
 
 std::vector<std::string> BaseVirtualMachine::get_all_ipv4(const SSHKeyProvider& key_provider)
 {
     std::vector<std::string> all_ipv4;
 
-    if (state == State::running)
+    if (current_state() == State::running)
     {
-        SSHSession session{ssh_hostname(), ssh_port(), ssh_username(), key_provider};
+        QString ip_a_output;
 
-        auto ip_a_output = QString::fromStdString(
-            mpu::run_in_ssh_session(session, "ip -brief -family inet address show scope global"));
+        try
+        {
+            SSHSession session{ssh_hostname(), ssh_port(), ssh_username(), key_provider};
 
-        QRegularExpression ipv4_re{QStringLiteral("([\\d\\.]+)\\/\\d+\\s*$"), QRegularExpression::MultilineOption};
+            ip_a_output = QString::fromStdString(
+                mpu::run_in_ssh_session(session, "ip -brief -family inet address show scope global"));
 
-        QRegularExpressionMatchIterator ip_it = ipv4_re.globalMatch(ip_a_output);
+            QRegularExpression ipv4_re{QStringLiteral("([\\d\\.]+)\\/\\d+\\s*$"), QRegularExpression::MultilineOption};
 
-        while (ip_it.hasNext())
-        {
-            auto ip_match = ip_it.next();
-            auto ip = ip_match.captured(1).toStdString();
+            QRegularExpressionMatchIterator ip_it = ipv4_re.globalMatch(ip_a_output);
 
-            all_ipv4.push_back(ip);
+            while (ip_it.hasNext())
+            {
+                auto ip_match = ip_it.next();
+                auto ip = ip_match.captured(1).toStdString();
+
+                all_ipv4.push_back(ip);
+            }
+        }
+        catch (const std::exception& e)
```

Comment:
I don't think we should sink all exceptions here. We're OK with SSH errors, but we'd want to know about `YouPCIsOnFireException` :wink: This would mean throwing a specific exception [here](https://github.com/canonical/multipass/blob/e1ac3898263ff735512177580e1a899d73a11aaf/include/multipass/ssh/throw_on_error.h#L39), [here](https://github.com/canonical/multipass/blob/e7913ee116068eaefa726336d8537b901a3ab37d/src/ssh/ssh_process.cpp#L67), and [here](https://github.com/canonical/multipass/blob/e7913ee116068eaefa726336d8537b901a3ab37d/src/ssh/ssh_process.cpp#L146). It could be a very general `SSHException` type. `ExitlessSSHProcessException` should probably inherit from it (we'd want to catch it here). I would just put both in the same header (e.g. ssh_exceptions.h).

There could be other things we'd want to ignore, but that's not the same as ignoring all cases. We'd have to revisit to catch them as needed. False positives are better than false negatives :smirk:

Reply:
Right. I created mp::SSHException and used it where necessary.